lib/charms/mongodb/v0/mongodb_tls.py

```diff
@@ -187,13 +209,12 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
         self.charm.delete_tls_certificate_from_workload()
         self.charm.push_tls_certificate_to_workload()
         self.charm.unit.status = MaintenanceStatus("enabling TLS")
-        self.charm.restart_mongod_service()
+        self.charm.restart_charm_services()
 
-        with MongoDBConnection(self.charm.mongodb_config) as mongo:
-            if not mongo.is_ready:
-                self.charm.unit.status = WaitingStatus("Waiting for MongoDB to start")
-            else:
-                self.charm.unit.status = ActiveStatus()
+        if not self.charm.is_db_service_ready():
+            self.charm.unit.status = WaitingStatus("Waiting for MongoDB to start")
+        else:
+            self.charm.unit.status = ActiveStatus()
```

Comment:
I think we should clear the status `Waiting for MongoDB to start` (if set) instead of setting ActiveStatus (this could override an existing status set by another component)